src/types-and-traits.rst

```diff
@@ -2190,6 +2190,25 @@ The :t:`type inference` algorithm uses :t:`type unification` to propagate known
 :t:`type inference root` being inferred. In the rules detailed below, a static
 error occurs when :t:`type unification` fails.
 
+:dp:`fls_6GrNr2izovRN`
+Performing :t:`type inference` may introduce a requirement that some :t:`type`
+must implement a :t:`trait`, or that a :t:`type` or :t:`lifetime` must outlive
+some other :t:`lifetime`. Such requirements are referred to as
+:dt:`[obligation]s` and are detailed in the inference rules below.
+
+:dp:`fls_9dSltJ6U98Fo`
+If insufficient :t:`type` information is available at the time an
+:t:`obligation` is introduced, it may be deferred to be resolved later.
+Any time new :t:`type` information is derived during :t:`type inference`, the
+tool shall attempt to resolve all outstanding :t:`[obligation]s` and propagate
```

Comment:
```suggestion
tool attempts to resolve all outstanding :t:`[obligation]s` and propagate
```
"shall" is for requirements aimed at the programmer.

Reply:
hmm, most of our "shall" requirements are actually checked by the tool it seems